react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+class NewsLetterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: ' '
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({
+      email: event.target.value
+    });
+  }
+
+  handleSubmit(event) {
+    console.log('state:', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label>
+          Email: <input type="text" onChange={this.handleChange} />
```

Comment:
Be sure to also pass a `value` prop to your `<input>` so that the component is fully "controlled".